mapie/tests/test_mondrian.py

```diff
@@ -0,0 +1,434 @@
+        "estimator": MapieClassifier,
+        "task": "classification",
+        "kwargs": {"conformity_score": RAPSConformityScore()}
+    },
+    "regression_residual_conformity": {
+        "estimator": MapieRegressor,
+        "task": "regression",
+        "kwargs": {"conformity_score": ResidualNormalisedScore()}
+    }
+}
+
+NON_VALID_CS_NAMES = list(NON_VALID_CS.keys())
+
+NON_VALID_MAPIE_ESTIMATORS = {
+    "calibration": {
+        "estimator": MapieCalibrator,
+        "task": "calibration",
+        "kwargs": {"method": "top_label", "random_state": 0}
+    },
+    "multi_label_recall_crc": {
+        "estimator": MapieMultiLabelClassifier,
+        "task": "multilabel_classification",
+        "kwargs": {"metric_control": "recall", "method": "crc"}
+    },
+    "multi_label_recall_rcps": {
+        "estimator": MapieMultiLabelClassifier,
+        "task": "multilabel_classification",
+        "kwargs": {"metric_control": "recall", "method": "rcps"},
+        "predict_kargs": {"delta": 0.01}
+    },
+    "multi_label_precision_ltt": {
+        "estimator": MapieMultiLabelClassifier,
+        "task": "multilabel_classification",
+        "kwargs": {"metric_control": "precision", "method": "ltt"},
+        "predict_kargs": {"delta": 0.01}
+    },
+    "mapie_quantile": {
+        "estimator": MapieQuantileRegressor,
+        "task": "regression",
+        "kwargs": {"method": "quantile"}
+    },
+    "mapie_time_series": {
+        "estimator": MapieTimeSeriesRegressor,
+        "task": "regression",
+        "kwargs": {"method": "quantile"}
+    }
+}
+
+NON_VALID_MAPIE_ESTIMATORS_NAMES = list(NON_VALID_MAPIE_ESTIMATORS.keys())
+
+TOY_DATASETS = {
+    "calibration": make_classification(
+        n_samples=1000, n_features=5, n_informative=5,
+        n_redundant=0, n_classes=10
+    ),
+    "classification": make_classification(
+        n_samples=1000, n_features=5, n_informative=5,
+        n_redundant=0, n_classes=10
+    ),
+    "multilabel_classification": make_multilabel_classification(
+        n_samples=1000, n_features=5, n_classes=5, allow_unlabeled=False
+    ),
+    "regression": make_regression(
+        n_samples=1000, n_features=5, n_informative=5
+    )
+
+}
+
+ML_MODELS = {
+    "calibration": LogisticRegression(),
+    "classification": LogisticRegression(),
+    "multilabel_classification": MultiOutputClassifier(
+                LogisticRegression(multi_class="multinomial")
+            ),
+    "regression": LinearRegression(),
+}
+
+
+@pytest.mark.parametrize("mapie_estimator_name", VALID_MAPIE_ESTIMATORS_NAMES)
+def test_valid_estimators_dont_fail(mapie_estimator_name):
+    """
+    Test that valid estimators don't fail"""
+    task_dict = VALID_MAPIE_ESTIMATORS[mapie_estimator_name]
+    mapie_estimator = task_dict["estimator"]
+    mapie_kwargs = task_dict["kwargs"]
+    task = task_dict["task"]
+    x, y = TOY_DATASETS[task]
+    y = np.abs(y)  # to avoid negative values with Gamma NCS
```

Comment:
What is the problem when y is negative? Does it not work? Does the end user need to know about this problem?

Reply:
problem linked to Gamma score, out of the scope of this PR